core/src/main/java/com/yahoo/gondola/core/Peer.java

```diff
@@ -648,7 +654,9 @@ public void run() {
                     String m = String.format("[%s-%d] Failed to send to %d: %s",
                             gondola.getHostId(), cmember.memberId, peerId,
                             e.getMessage());
+                    
                     if ("Socket closed".equals(e.getMessage())
+                            || matchesMessage(e, ".*Read end dead.*")
```

Comment:
yes agreed. I would like to either find a way to avoid this line or replace pipedinputstream.